sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -193,16 +193,44 @@ class Analyzer(
     }
 
     def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+      case a if !a.childrenResolved => a // be sure all of the children are resolved.
       case a: Cube =>
         GroupingSets(bitmasks(a), a.groupByExprs, a.child, a.aggregations)
       case a: Rollup =>
         GroupingSets(bitmasks(a), a.groupByExprs, a.child, a.aggregations)
       case x: GroupingSets =>
         val gid = AttributeReference(VirtualColumn.groupingIdName, IntegerType, false)()
+
+        // the complex expression (non-attribute expressions) in the GROUP BY keys
+        val nonAttributeGroupByExpression = new ArrayBuffer[Alias]()
+        val groupByExprPairs = x.groupByExprs.map(_ match {
+          case e: NamedExpression => (e, e)
+          case other => {
+            val alias = Alias(other, other.toString)()
+            nonAttributeGroupByExpression += alias
+            (other, alias.toAttribute) // (Aliased complex expression, the associated attribute)
+          }
+        })
```

Comment:
Probably we want to keep the order group by expressions in the Aggregate operator, let's just keep it as it used to be.